src/start.ts

```diff
@@ -0,0 +1,83 @@
+import {getInput, info} from '@actions/core'
+import {exec} from '@actions/exec'
+import {join} from 'path'
+import {tmpdir} from 'os'
+import {promises} from 'fs'
+import {wait} from './wait'
+import optionMappingJson from './option-mapping.json'
+import {stopContainer} from './stop-container'
+
+const LOG_FILE = '/srv/sauce-connect.log'
+const PID_FILE = '/srv/sauce-connect.pid'
+const READY_FILE = '/opt/sauce-connect-action/sc.ready'
+
+type OptionMapping = {
+    actionOption: string
+    dockerOption: string
+    required?: boolean
+    flag?: boolean
+}
+const optionMappings: OptionMapping[] = optionMappingJson
+
+function buildOptions(): string[] {
+    const params = [
+        `--logfile=${LOG_FILE}`,
+        `--pidfile=${PID_FILE}`,
+        `--readyfile=${READY_FILE}`,
+        `--verbose`
+    ]
+
+    for (const optionMapping of optionMappings) {
+        const input = getInput(optionMapping.actionOption, {
+            required: optionMapping.required
+        })
+
+        if (input === '') {
+            // user input nothing for this option
+        } else if (optionMapping.flag) {
+            // for flag options like --doctor option
+            params.push(`--${optionMapping.dockerOption}`)
+        } else {
+            params.push(`--${optionMapping.dockerOption}=${input}`)
+        }
+    }
+    return params
+}
+
+export async function start(): Promise<string> {
```

Comment:
Should we call this `startContainer` given the other method is `stopContainer`

Reply:
Yep will update